darwin/exporter/formats/mask.py

```diff
@@ -1,58 +1,390 @@
+        The annotation file that the annotations belong to.
+    height: int
+        The height of the image.
+    width: int
+        The width of the image.
+
+    Returns
+    -------
+    Tuple[List[Exception], Image, dt.MaskTypes.CategoryList, dt.MaskTypes.ColoursDict]
+    """
+
+    errors: List[Exception] = []
+
+    for a in annotations:
+        try:
+            if isinstance(a, dt.VideoAnnotation):
+                print(f"Skipping video annotation from file {annotation_file.filename}")
+                continue
+
+            cat = a.annotation_class.name
+            if not cat in categories:
+                categories.append(cat)
+
+            if a.annotation_class.annotation_type == "polygon":
+                polygon = a.data["path"]
+            elif a.annotation_class.annotation_type == "complex_polygon":
+                polygon = a.data["paths"]
+            else:
+                raise ValueError(f"Unknown annotation type {a.annotation_class.annotation_type}")
+            sequence = convert_polygons_to_sequences(polygon, height=height, width=width)
+
+            colour_to_draw = get_or_generate_colour(cat, colours)
+            mask = draw_polygon(mask, sequence, colour_to_draw)
+
+            if cat not in colours:
+                colours[cat] = colour_to_draw
+
+        except Exception as e:
+            errors.append(e)
+            continue
+
+    # It's not necessary to return the mask, it's modified in place, but it's more explicit
+    return errors, mask, categories, colours
+
+
+def render_raster(
+    mask: NDArray,
+    colours: dt.MaskTypes.ColoursDict,
+    categories: dt.MaskTypes.CategoryList,
+    annotations: List[dt.AnnotationLike],
+    annotation_file: dt.AnnotationFile,
+    height: int,
+    width: int,
+) -> dt.MaskTypes.RendererReturn:
+    """
+    Renders the raster layers in the given annotations onto the given mask.
+
+    Parameters
+    ----------
+    mask: NDArray
+        The mask to render the polygons onto.
+    annotations: List[dt.AnnotationLike]
+        A list of annotations to be rendered.
+    annotation_file: dt.AnnotationFile
+        The annotation file that the annotations belong to.
+    height: int
+        The height of the image.
+    width: int
+        The width of the image.
+
+    Returns
+    -------
+    Tuple[List[Exception], Image, dt.MaskTypes.CategoryList, dt.MaskTypes.ColoursDict]
+    """
+    errors: List[Exception] = []
+
+    mask_annotations: List[dt.AnnotationMask] = []
+    raster_layer: Optional[dt.RasterLayer] = None
+
+    mask_lookup: Dict[str, dt.AnnotationMask] = dict()
+
+    for a in annotations:
+        if isinstance(a, dt.VideoAnnotation):
+            continue
+
+        data = a.data
+
+        if "mask" in data:
+            new_mask = dt.AnnotationMask(
+                id=data["id"],
+                name=data["name"],
+                slot_names=a.slot_names,
+            )
+            new_mask.validate()
```

Comment:
can we catch this like the other errors or will that present other issues?

Reply:
Sure can.